packages/dev/core/src/scene.ts

```diff
@@ -2354,6 +2354,15 @@ export class Scene implements IAnimatable, IClipPlanesHolder, IAssetContainer {
             }
         }
 
+        // Effect layers
+        if (this.effectLayers) {
+            for (const effectLayer of this.effectLayers) {
+                if (!effectLayer.isLayerReady()) {
+                    isReady = false;

```

Comment:
nit - you can break the loop here, no?

Reply:
No, we want to loop over all the layers, to start compiling the shaders (if any) as soon as possible. If we break early, these shaders will not start to be compiled until the next time `isReady` is called (it is even possible that several calls to `isReady` will be necessary before all the shaders start to be compiled), which would delay the moment when the scene is ready.